src/Database/TypeMap.php

```diff
@@ -69,7 +69,7 @@ public function __construct(array $defaults = [])
      * This method will replace all the existing default mappings with the ones provided.
      * To add into the mappings use `addDefaults()`.
      *
-     * @param array<string, string> $defaults Associative array where keys are field names and values
+     * @param array<string> $defaults Associative array where keys are field names and values
```

Comment:
Shouldnt we then use 
```suggestion
     * @param array<int|string, string> $defaults Associative array where keys are field names and values
```
to make clear we looked at the keys here.
It otherwise looks more like `list<string>` or we didnt yet check the keys.

As for the description: It now doesnt really fit anymore to the types
Do we plan to fix this up for 5.x then? Or should we just change the text and remove the assoc part or clarify the non assoc one?